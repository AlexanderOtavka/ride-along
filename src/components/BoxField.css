/**
 * BoxField.css
 *
 * Created by Zander Otavka on 6/4/17.
 * Copyright (C) 2016  Grinnell AppDev.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

.BoxField {
  flex-grow: 1;
  height: 48px;
  background: rgba(242, 242, 242, 0.3);
  border: none;
  padding: 0 16px;
  font-size: 18px;
  color: white;
  font-weight: 300;
  outline: none;
  text-align: left;
}

.BoxField:focus {
  background-color: #F2F2F2;
  color: #4F4F4F;
}

.BoxField::placeholder {
  color: white;
}

.BoxField:focus::placeholder {
  color: #4F4F4F;
}
