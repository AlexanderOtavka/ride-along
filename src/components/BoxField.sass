/**
 * BoxField.sass
 *
 * Created by Zander Otavka on 6/4/17.
 * Copyright (C) 2016  Grinnell AppDev.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

@import "style/colors"
@import "style/effects"
@import "style/layout"

.boxField
  position: relative
  flex-grow: 1
  z-index: 0

.backdrop
  +fullbleed

  background-color: $off-white
  opacity: 0.3
  z-index: -1
  transition: opacity 400ms
  will-change: opacity

  .input:focus ~ &
    opacity: 1
    box-shadow: $shadow1

.rippleWrapper
  overflow: hidden

.ripple
  background-color: $off-white

.input
  width: 100%
  height: 48px
  background: none
  border: none
  padding: 0 16px
  font-size: 18px
  color: white
  font-weight: 300
  outline: none
  text-align: left

  &::placeholder
    color: white

  &:focus
    color: $grey2

    &::placeholder
      color: $grey2
