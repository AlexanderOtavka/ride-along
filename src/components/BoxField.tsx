/**
 * @file BoxField.tsx
 *
 * Created by Zander Otavka on 6/4/17.
 * Copyright (C) 2016  Grinnell AppDev.
 *
 * @license
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

import React from "react"
import { FormField } from "react-form"
import classnames from "classnames"

import styles from "./BoxField.sass"

interface Props extends React.HTMLProps<HTMLInputElement> {
  isButton?: boolean
  field: string
}

export default function BoxField({
  field,
  isButton = false,
  className,
  placeholder,
  value,
  ...props,
}: Props) {
  const classes = classnames(styles.boxField, className)

  return isButton
    ? <input {...props} className={classes} type="submit" value={placeholder} />
    : <FormField field={field}>
        {({ getValue, setValue, setTouched }: any) =>
          <input
            {...props}
            className={classes}
            placeholder={placeholder}
            value={getValue("")}
            onChange={ev => setValue(ev.target.value)}
            onBlur={ev => setTouched()}
          />}
      </FormField>
}
