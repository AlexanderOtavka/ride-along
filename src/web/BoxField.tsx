/**
 * @file BoxField.tsx
 *
 * Created by Zander Otavka on 6/4/17.
 * Copyright (C) 2016  Grinnell AppDev.
 *
 * @license
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

import React from "react"
import { FormField } from "react-form"
import ripple, { RippleTheme } from "react-toolbox/lib/ripple/Ripple"
import classnames from "classnames"

import rippleTheme from "react-toolbox/lib/ripple/theme.css"
import styles from "./BoxField.sass"

export interface Props extends React.HTMLProps<HTMLInputElement> {
  field: string
  type?: "text" | "date" | "submit"
  theme?: RippleTheme
  rootProps?: React.HTMLProps<HTMLDivElement>
}

const withRipple = ripple({
  theme: {
    ...rippleTheme,
    ripple: classnames(styles.ripple, rippleTheme.ripple),
    rippleWrapper: classnames(styles.rippleWrapper, rippleTheme.rippleWrapper),
  },
})

function BoxField({
  field,
  type = "text",
  rootProps,
  placeholder,
  value,
  theme,
  children,
  onMouseDown,
  onTouchStart,
  ...props,
}: Props) {
  const handleMouseDown = (ev: React.MouseEvent<HTMLInputElement>) => {
    if (onMouseDown && document.activeElement !== ev.currentTarget) {
      onMouseDown(ev)
    }
  }

  const handleTouchStart = (ev: React.TouchEvent<HTMLInputElement>) => {
    if (onTouchStart && document.activeElement !== ev.currentTarget) {
      onTouchStart(ev)
    }
  }

  // Autofocus is broken in preact for some reason, so we have to do this hack
  const autofocusRefHandler = (el: HTMLInputElement) => {
    if (el && props.autoFocus) {
      el.focus()
    }
  }

  return (
    <div
      {...rootProps}
      className={classnames(styles.boxField, rootProps && rootProps.className)}
    >
      {type === "submit" ? (
        <input
          {...props}
          ref={autofocusRefHandler}
          className={classnames(styles.input, props.className)}
          type="submit"
          value={placeholder}
          onMouseDown={handleMouseDown}
          onTouchStart={handleTouchStart}
        />
      ) : (
        <FormField field={field}>
          {({ getValue, setValue, setTouched }: any) => (
            <input
              {...props}
              ref={autofocusRefHandler}
              type={type}
              className={classnames(styles.input, props.className)}
              placeholder={placeholder}
              value={getValue("")}
              onChange={ev => {
                setValue(ev.currentTarget.value)
                if (props.onChange) {
                  props.onChange(ev)
                }
              }}
              onBlur={ev => {
                setTouched()
                if (props.onBlur) {
                  props.onBlur(ev)
                }
              }}
              onMouseDown={handleMouseDown}
              onTouchStart={handleTouchStart}
            />
          )}
        </FormField>
      )}
      <div className={styles.backdrop} />
      {children}
    </div>
  )
}

export default withRipple(BoxField)
