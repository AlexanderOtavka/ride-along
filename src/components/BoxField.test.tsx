/**
 * @file BoxField.test.tsx
 *
 * Created by Zander Otavka on 6/6/17.
 * Copyright (C) 2016  Grinnell AppDev.
 *
 * @license
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

import React from "react"
import renderer from "react-test-renderer"

import BoxField from "./BoxField"

it("has the correct className when rendered without props", () => {
  const component = renderer.create(<BoxField />)

  expect(component.toJSON()).toMatchSnapshot()
})

it("has the correct className when rendered with props", () => {
  const component = renderer.create(<BoxField className="foo" />)

  expect(component.toJSON()).toMatchSnapshot()
})
