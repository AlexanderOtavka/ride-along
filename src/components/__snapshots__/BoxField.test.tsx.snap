// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has the correct className when rendered with props 1`] = `
<input
  className="boxField foo"
/>
`;

exports[`has the correct className when rendered without props 1`] = `
<input
  className="boxField"
/>
`;
